frontend/components/service/serviceDetailsCard.js

```diff
@@ -1,256 +1,240 @@
-    setRemainEmployeeList(
-      employeeList.filter(
-        ({ firstname: val1 }) =>
-          !serviceEmployList.some(({ firstname: val2 }) => val2 === val1)
-      )
-    );
-  };
+    const handleAddEmployeeCheck = (val, employee) => {
+        if (val) {
+            setAddEmployeeCheckList([...addEmployeeCheckList, employee]);
+        } else {
+            setAddEmployeeCheckList(
+                addEmployeeCheckList.filter(
+                    (emp) => emp.id != employee.id
+                )
+            );
+        }
+    };
 
-  const handleAddSelected = () => {
-    setServiceEmployList([...serviceEmployList, ...addEmployeeCheckList]);
-    setServiceEmployeeDialog(false);
-    setAddEmployeeCheckList([]);
-  };
+    const handleDeleteEmployeeCheck = (val, employee) => {
+        if (val) {
+            setDeleteEmployeeCheckList([...deleteEmployeeCheckList, employee]);
+        } else {
+            setDeleteEmployeeCheckList(
+                deleteEmployeeCheckList.filter(
+                    (emp) => emp.id != employee.id
+                )
+            );
+        }
+    };
+    const handleAddEmployee = () => {
+        setServiceEmployeeDialog(true);
+        setRemainEmployeeList(
+            employeeList.filter(
+                ({id: val1}) =>
+                    !serviceEmployList.some(({id: val2}) => val2 === val1)
+            )
+        );
+    };
 
-  const handleDeleteEmployee = () => {
-    setServiceEmployList(
-      serviceEmployList.filter(
-        ({ firstname: val1 }) =>
-          !deleteEmployeeCheckList.some(({ firstname: val2 }) => val2 === val1)
-      )
-    );
-    setDeleteEmployeeCheckList([]);
-  };
-  const handleSaveService = () => {
-    let data = {
-      serviceCode: barcode,
-      name: 'service 1',
-      description: description,
-      treatment_type: 'type 1',
-      duration: 60 * 3600000,
-      price: 120,
-      barcode: barcode,
-      sms_description: 'sms description 1',
+    const handleAddSelected = () => {
+        setServiceEmployList([...serviceEmployList, ...addEmployeeCheckList]);
+        setServiceEmployeeDialog(false);
+        setAddEmployeeCheckList([]);
     };
-    editHandle(data, item.id);
-  };
 
-  const handleAddDurationPrice = () => {
-    setDurationPriceList([...durationPriceList, { price: 0, duration: 0.5 }]);
-  };
-  const handleRemoveDurationPrice = (index) => {
-    setDurationPriceList([
-      ...durationPriceList.slice(0, index),
-      ...durationPriceList.slice(index + 1),
-    ]);
-  };
+    const handleDeleteEmployee = () => {
+        setServiceEmployList(
+            serviceEmployList.filter(
+                ({id: val1}) =>
+                    !deleteEmployeeCheckList.some(({id: val2}) => val2 === val1)
+            )
+        );
+        setDeleteEmployeeCheckList([]);
+    };
+    const handleSaveService = () => {
+        let durationTemp = durationPriceList.map(d => (
```

Comment:
I suggest inline the next two lines. See https://refactoring.com/catalog/inlineVariable.html